datafusion/physical-expr/src/physical_expr.rs

```diff
@@ -104,6 +105,14 @@ pub trait PhysicalExpr: Send + Sync + Display + Debug + PartialEq<dyn Any> {
             "Not implemented for {self}"
         )))
     }
+
+    fn dyn_hash(&self, _state: &mut dyn Hasher);
```

Comment:
I think we should add some comments here to help anyone who tries to implement `PhysicalExpr`

Perhaps we can follow the model of the documentation on 

https://docs.rs/datafusion-expr/26.0.0/datafusion_expr/logical_plan/trait.UserDefinedLogicalNode.html#tymethod.dyn_hash